### PR TITLE
[Feature/extensions] Renaming to match merge to main branch

### PR DESCRIPTION
**Pull request by ryanbogan**
Signed-off-by: Ryan Bogan <rbogan@amazon.com>

### Description
Renames classes, methods, and variables to match the merge to main branch

Companion PR on opensearch-sdk-java: https://github.com/opensearch-project/opensearch-sdk-java/pull/276

### Check List
- [X] New functionality includes testing.
  - [X] All tests pass
- [X] New functionality has been documented.
  - [X] New functionality has javadoc added
- [X] Commits are signed per the DCO using --signoff
- [X] Commit changes are listed out in CHANGELOG.md file (See: [Changelog](../blob/main/CONTRIBUTING.md#changelog))

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by ryanbogan** [comment]
Looks like main needs to be merged into feature/extensions before this can be merged.

**Review by joshpalis — APPROVED** [review]

File: server/src/main/java/org/opensearch/extensions/ExtensionsManager.java
Comment (ryanbogan) [inline server/src/main/java/org/opensearch/extensions/ExtensionsManager.java:415]: This code is really awkward, but for some reason there is an `InterruptedException` that is not caught by the `Exception` class inherently, so this is the only way I could get it to work.  I'm open for suggestions if anyone else can come up with a more elegant solution.

**Comment by saratvemulapalli** [comment]
@ryanbogan looks like we did rename quite a few, wouldn't this need a companion PR on SDK ?

**Comment by ryanbogan** [comment]
> @ryanbogan looks like we did rename quite a few, wouldn't this need a companion PR on SDK ?

Right I forgot about that.  I'll start on that now.

File: server/src/main/java/org/opensearch/common/settings/IndexScopedSettings.java
Comment (owaiskazi19) [inline server/src/main/java/org/opensearch/common/settings/IndexScopedSettings.java:255]: Why are we making this change?

Comment (reta) [reply server/src/main/java/org/opensearch/common/settings/IndexScopedSettings.java:255]: `super.registerSetting(setting)` calls `validateSettingKey`

**Review by owaiskazi19 — APPROVED** [review]

File: server/src/main/java/org/opensearch/common/settings/IndexScopedSettings.java
Comment (reta) [inline server/src/main/java/org/opensearch/common/settings/IndexScopedSettings.java:254]: I think you don't need override anymore after https://github.com/opensearch-project/OpenSearch/pull/5362/files#diff-edf64b791aed5932da1f53fa264e42e98bd473f585ceed12892b66cda79c3fd7R147 ?

File: server/src/main/java/org/opensearch/extensions/ExtensionsManager.java
Comment (reta) [reply server/src/main/java/org/opensearch/extensions/ExtensionsManager.java:409]: :o this should not be possible, do you have a way to reproduce that?

File: server/src/main/java/org/opensearch/extensions/ExtensionsManager.java
Comment (reta) [inline server/src/main/java/org/opensearch/extensions/ExtensionsManager.java:366]: Sorry, I missed that previously, could we replace lath with `CompletableFuture` instead?

File: server/src/main/java/org/opensearch/common/settings/IndexScopedSettings.java
Comment (ryanbogan) [reply server/src/main/java/org/opensearch/common/settings/IndexScopedSettings.java:254]: @reta As in we can just delete the method entirely?

**Review by saratvemulapalli — APPROVED** [review]
Lets wait for @reta's ack before we merge. Thanks @ryanbogan for the change.

**Comment by reta** [comment]
@saratvemulapalli LGTM, thank you, I do have few comments but no need to block the merge, could be addressed later

**Pull request merged**
